x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/runtimefields/RuntimeFieldsFeatureSetUsage.java

```diff
@@ -89,6 +98,21 @@ public static RuntimeFieldsFeatureSetUsage fromMetadata(Iterable<IndexMetadata>
         return new RuntimeFieldsFeatureSetUsage(Collections.unmodifiableList(runtimeFieldStats));
     }
 
+    @SuppressWarnings("unchecked")
+    private static void findFieldNames(Map<String, Object> mappings, String path, Consumer<String> fieldNameConsumer) {
+        Object o = mappings.get("properties");
```

Comment:
It would be really great to reuse MappingVisitor here, as it does the right thing, only a shame that if does a fundamentally different thing, as in it does not allow you to consume the name, but rather the content of the field mappings.

Reply:
I've rewritten to use MappingVisitor, which handles objects and multifields.